### PR TITLE
Enable antsibull-changelog and add GitHub workflow for changelog validation

### PR DESCRIPTION
**Pull request by alinabuzachis**
Enable antsibull-changelog and add GitHub workflow for changelog validation

### CODE REVIEW

File: changelogs/fragments/update_dependencies.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+breaking_changes:
```
Comment (gravesm) [inline changelogs/fragments/update_dependencies.yml:1]: This one also should not be a breaking change. There should be no change from the end user's standpoint.

Comment (alinabuzachis) [reply changelogs/fragments/update_dependencies.yml]: I thought users were constrained to update their dependency versions.

File: changelogs/fragments/update_dependencies.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+trivial:
```
Comment (abikouo) [inline changelogs/fragments/update_dependencies.yml:1]:
```suggestion
minor_changes:
```

**Review by abikouo — APPROVED** [review]

**Review by hakbailey — APPROVED** [review]

File: changelogs/fragments/update_dependencies.yml (hunk 1 above)
Comment (gravesm) [reply changelogs/fragments/update_dependencies.yml]: Users don't have to update any dependencies. We're distributing a binary that has all the dependencies built into it.  As far as the user is concerned, it's just a new version of the provider that works just like the old version.

**Review by gravesm — APPROVED** [review]

**Pull request merged**